public/components/Flyout/flyouts/createDetector.js

```diff
@@ -0,0 +1,833 @@
+      fieldRangeStart:
+        newValue.where.fieldRangeStart === undefined ? 0 : newValue.where.fieldRangeStart,
+      fieldValue: newValue.where.fieldValue === undefined ? '' : newValue.where.fieldValue,
+      operator: newValue.where.operator === undefined ? 'is' : newValue.where.operator,
+    },
+  };
+  let filterQuery = formikToWhereClause(search);
+  let adFilterQuery = '';
+  if (filterQuery) {
+    adFilterQuery = { bool: { filter: [filterQuery] } };
+  } else {
+    adFilterQuery = { match_all: { boost: 1.0 } };
+  }
+  context.adConfigs.filter_query = adFilterQuery;
+  context.adConfigs.name = newValue.name;
+  context.adConfigs.description = newValue.description;
+  context.adConfigs.time_field = newValue.time_field;
+  context.adConfigs.inidices = newValue.indices;
+  context.adConfigs.window_delay = { period: { interval: newValue.window_delay, unit: 'MINUTES' } };
+  context.adConfigs.detection_interval = {
+    period: { interval: newValue.detection_interval, unit: 'MINUTES' },
+  };
+  context.queriesForOverview.where = search.where;
+  context.queriesForOverview.filter_query = adFilterQuery;
+  const configs = context.adConfigs;
+  const httpClient = context.httpClient;
+  try {
+    const response = await httpClient.post('../api/alerting/detectors/_validate', {
+      configs,
+    });
+    let resp = _.get(response, 'data.response');
+    const {
+      data: {
+        ok,
+        response: { _id },
+      },
+    } = response;
+    if (ok) {
+      context.setFlyout(null);
+      context.renderFlyout(configs, resp, context.queriesForOverview);
+    }
+  } catch (err) {
+    if (typeof err === 'string') throw err;
+    throw 'There was a problem validating the configurations';
+  }
+}
+
+const ConfigCell = (props) => {
+  return (
+    <FixedWidthRow label={props.title}>
+      <EuiText>
+        <p className="enabled">{props.description}</p>
+      </EuiText>
+    </FixedWidthRow>
+  );
+};
+
+const validationErrorCallOut = (failures, suggestedChanges, field) => {
+  let message;
+  for (let [key, value] of Object.entries(failures)) {
+    if (key === 'duplicates' && field === 'name') {
+      message = 'Detector name is a duplicate';
+    } else if (key === 'missing' && value[0] === field) {
+      message = 'This field is required';
+    } else if ((key === 'regex' && field === 'name') || (key === 'format' && field === 'name')) {
+      message = 'Valid characters are a-z, A-Z, 0-9, -(hyphen) and _(underscore)';
+    }
+  }
+  for (let [key, value] of Object.entries(suggestedChanges)) {
+    if (key === 'window_delay' && field === 'window_delay') {
+      message = 'Window delay should be at least ' + value[0] + ' minutes';
+    } else if (key === 'filter_query' && field === 'filter_query') {
+      message = value[0];
+    } else if (key === 'detectionIntervalMax' && field === 'detection_interval') {
+      message = value;
+    }
+  }
```

Comment:
Looks like `message` could get overwritten here if there was multiple failures and/or suggested changes. How do we propagate all to the user? This may also need to be discussed with UX, where a callout may not be the right solution here if there's several (2+) things the user should be notified of.